libraries/common/subscriptions/router.js

```diff
@@ -64,10 +69,30 @@ export default function router(subscribe) {
       }
     }
 
-    // Check for a redirect url and change location if one is found.
-    const redirect = redirects.get(location);
+    // Check for a redirect and change location if one is found.
+    let redirect = redirects.get(location);
 
     if (redirect) {
+      // Check if a redirect handler was assigned and execute it.
+      if (typeof redirect === 'function' || redirect instanceof Promise) {
```

Comment:
Please make this a guard clause. We don't wanna have this massive if / else blocks in the code.

Reply:
i don't think that is possible here, since the code execution only breaks if the redirect handler was rejected.